src/server/game/Entities/Totem/Totem.cpp

```diff
@@ -54,19 +55,22 @@ void Totem::Update(uint32 time)
 void Totem::InitStats(uint32 duration)
 {
     // client requires SMSG_TOTEM_CREATED to be sent before adding to world and before removing old totem
-    if (GetOwner()->GetTypeId() == TYPEID_PLAYER
-            && m_Properties->Slot >= SUMMON_SLOT_TOTEM
-            && m_Properties->Slot < MAX_TOTEM_SLOT)
+    if (Player* owner = GetOwner()->ToPlayer())
     {
-        WorldPacket data(SMSG_TOTEM_CREATED, 1 + 8 + 4 + 4);
-        data << uint8(m_Properties->Slot - 1);
-        data << uint64(GetGUID());
-        data << uint32(duration);
-        data << uint32(GetUInt32Value(UNIT_CREATED_BY_SPELL));
-        GetOwner()->ToPlayer()->SendDirectMessage(&data);
+        uint32 slot = m_Properties->Slot;
+        if (slot >= SUMMON_SLOT_TOTEM_FIRE && slot < MAX_TOTEM_SLOT)
+        {
+            WorldPacket data(SMSG_TOTEM_CREATED, 1 + 8 + 4 + 4);
```

Comment:
the perfect finish for this PR... port the new packet structure 💃

Reply:
also, wouldnt this fit better in `JustAppeared` ?

Reply:
or maybe not... that would imply to move it to AI... any thoughts @Treeston @Shauren ?

Reply:
AI sending packets gives me twitches. Why do you want it to be in AI?

Reply:
so the packet is sent when the totem is actually visible... makes sense?
let me lookup for any packet log to clarify what the order should be

Reply:
umm realizing now that maybe an override to AddToWorld could be better

Reply:
am I using this review to randomly post my thoughts?..... maybe

Reply:
nvm all this, the packet needs some thought, keep going